dqc_us_rules/dqc_us_0014.py

```diff
@@ -0,0 +1,92 @@
+# (c) Copyright 2016, XBRL US Inc. All rights reserved.
+# See license.md for license information.
+# See PatentNotice.md for patent infringement notice.
+import os
+
+from .util import facts, messages, neg_num
+import decimal
+
+
+_CODE_NAME = 'DQC.US.0014'
+_RULE_VERSION = '1.1'
+_DEFAULT_CONCEPTS_FILE = os.path.join(
+    os.path.dirname(__file__),
+    'resources',
+    'DQC_US_0014',
+    'dqc_14_concepts.csv'
+)
+
+
+def run_negative_numbers_no_dimensions(val):
+    """
+    Run the list of facts against our negative number check for facts with
+    no dimensions and add errors for the hits in the various lists.
+
+    :param val: The validation object which carries the validation information,
+        including the ModelXBRL
+    :return: No direct return but throws errors for facts matching the blacklist
+    :rtype: None
+    """
+    blacklist_dict = neg_num.concept_map_from_csv(_DEFAULT_CONCEPTS_FILE)
+    blacklist_facts = filter_negative_number_no_dimensions_facts(val, blacklist_dict.keys())
+    for fact in blacklist_facts:
+        index_key = blacklist_dict[fact.qname.localName]
+        val.modelXbrl.error(
+            '{base_key}.{extension_key}'.format(
+                base_key=_CODE_NAME, extension_key=index_key
+            ),
+            messages.get_message(_CODE_NAME, str(index_key)),
+            concept=fact.concept.label(),
+            modelObject=fact,
+            ruleVersion=_RULE_VERSION
+        )
+
+
+def filter_negative_number_no_dimensions_facts(val, blacklist_concepts):
+    """
+    Checks the numeric negative value dimensionless facts in the provided ModelXBRL instance
+    against the rule dictionary and returns those which meet the conditions of the blacklist.
+
+    :param val: The validatino object which carries the validation information,
```

Comment:
Actually it just looks like 2890 needs to be merged into this ticket and the target.